src/health_misinfo_shared/claim_format_checker.py

```diff
@@ -0,0 +1,37 @@
+from pydantic import BaseModel, ValidationError
+
+
+class LabelsModel(BaseModel):
+    understandability: str
+    type_of_claim: str
+    type_of_medical_claim: str
+    support: str
+    harm: str
+
+
+class ClaimModel(BaseModel):
+    claim: str
+    original_text: str
+    labels: LabelsModel
+
+
+def assert_output_json_format(output: dict) -> bool:
+    try:
+        ClaimModel(**output)
+    except ValidationError:
+        print(output)
+        return False
+    return True
+
+
+def insert_missing_key_as_null(output: dict) -> dict:
+    # fill in data if it is missing
+    if "original_text" not in output.keys():
+        output["original_text"] = "not found"
+    labels = output.get("labels", {})
+    # missing labels will be filled in as blank
+    for k in list(LabelsModel.model_fields):
+        if k not in labels.keys():
+            labels[k] = ""
+    output["labels"] = labels
+    return output
```

Comment:
This is the sort of thing pydantic can do for you. E.g.

```python
class LabelsModel(BaseModel):
    understandability: str = ""
    type_of_claim: str = ""
    type_of_medical_claim: str = ""
    support: str = ""
    harm: str = ""


class ClaimModel(BaseModel):
    claim: str
    original_text: str = "not found"
    labels: LabelsModel = LabelsModel()


candidate_list = {"claim": "the claim"}
candidate_list = ClaimModel(**candidate_list).dict()
print(candidate_list)
# {'claim': 'the claim', 'original_text': 'not found', 'labels': {'understandability': '', 'type_of_claim': '', 'type_of_medical_claim': '', 'support': '', 'harm': ''}}
```

Reply:
I might be wrong here but I think maybe if I do this, it would always pass the validation and therefore we wouldn't be able to make the second and third attempts?

Reply:
That’s right – but you could have a separate `ClaimModel` and `StrictClaimModel` for this purpose, where `ClaimModel` has defaults, and `StrictClaimModel` doesn’t.

I hoped pydantic might have a strict/non-strict mode for doing this sort of thing (which might be neater), but its strict mode is about type coercion.

Reply:
Ahhh got you. Attempting now, but with the caveat it takes a lot of attempts to get the errors which would trigger this to occur.

Reply:
> with the caveat it takes a lot of attempts to get the errors which would trigger this to occur.

Yeah, that’s fair… But tbh I think that’s an argument not to bother with the auto-populated fields thing, and instead just give up after n attempts (and maybe log that it happened, so we have an idea of how often this occurs).

Reply:
struggling to get this one to work so I'm going to leave it for now as it's not a high priority, but thank you!